src/core/server/audit_trail/types.ts

```diff
@@ -19,17 +19,78 @@
 import { KibanaRequest } from '../http';
 
 /**
- * Event to audit.
+ * Audit event using ECS format: https://www.elastic.co/guide/en/ecs/current/index.html
  * @public
  *
  * @remarks
  * Not a complete interface.
  */
-export interface AuditableEvent {
+export interface AuditEvent {
   message: string;
-  type: string;
+  event: {
+    action: string;
+    category?: EventCategory | EventCategory[];
+    type?: EventType | EventType[];
+    outcome?: EventOutcome;
+    module?: string;
+    dataset?: string;
+  };
+  user: {
+    name: string;
+    email?: string;
+    full_name?: string;
+    hash?: string;
+  };
+  kibana?: {
+    session_id?: string;
+    user_roles?: string[];
```

Comment:
Yeh, so I originally had the `roles` property in the `user` object like you suggested but their feedback was to move it into `kibana` object so that it doesn't clash with future fields they introduce to the schema. 

Let's see what they say to your proposal

Reply:
Looks like elastic/ecs#915 is getting some traction. There's a discussion ongoing, and a PR was raised to add support for this field: https://github.com/elastic/ecs/pull/917